SEFramework/SEFramework/Image/SnrImage.h

```diff
@@ -0,0 +1,87 @@
+/** Copyright © 2019 Université de Genève, LMU Munich - Faculty of Physics, IAP-CNRS/Sorbonne Université
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 3.0 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+/*
+ * SnrImage.h
+ *
+ *  Created on: May 24, 2019
+ *      Author: mkuemmel
+ */
+
+#ifndef _SEFRAMEWORK_IMAGE_SNRIMAGE_H_
+#define _SEFRAMEWORK_IMAGE_SNRIMAGE_H_
+
+#include <memory>
+
+#include "SEFramework/Image/Image.h"
+#include "SEFramework/Image/ImageBase.h"
+#include "SEFramework/Image/ConstantImage.h"
+
+namespace SourceXtractor {
+
+/**
+ * @class SnrImage
+ * @brief Used to subtract a constant value from an Image
+ *
+ */
+template <typename T>
```

Comment:
This should be implemented as a ProcessedImage (see examples at the end of ProcessedImage.h).

Reply:
Also that one should be settled as of now.